engine/src/cmd/unit_generic.cpp

```diff
@@ -1255,36 +1262,20 @@ void Unit::DamageRandSys(float dam, const Vector &vec, float randnum, float degr
         return;
     }
     if (degrees >= 35 && degrees < 60) {
-        //DAMAGE FUEL
-        static float fuel_damage_prob = 1.f
-                - XMLSupport::parse_float(vs_config->getVariable("physics", "fuel_damage_prob", ".25"));
-        static float warpenergy_damage_prob = fuel_damage_prob
-                - XMLSupport::parse_float(vs_config->getVariable("physics",
-                        "warpenergy_damage_prob",
-                        "0.05"));
-        static float ab_damage_prob = warpenergy_damage_prob
-                - XMLSupport::parse_float(vs_config->getVariable("physics", "ab_damage_prob", ".2"));
-        static float cargovolume_damage_prob = ab_damage_prob
-                - XMLSupport::parse_float(vs_config->getVariable("physics",
-                        "cargovolume_damage_prob",
-                        ".15"));
-        static float upgradevolume_damage_prob = cargovolume_damage_prob
-                - XMLSupport::parse_float(vs_config->getVariable("physics",
-                        "upgradevolume_damage_prob",
-                        ".1"));
-        static float cargo_damage_prob = upgradevolume_damage_prob
-                - XMLSupport::parse_float(vs_config->getVariable("physics", "cargo_damage_prob", "1"));
-        if (randnum >= fuel_damage_prob) {
-            fuel.Damage();
-        } else if (randnum >= warpenergy_damage_prob) {
-            ftl_energy.Damage();
-        } else if (randnum >= ab_damage_prob) {
-            this->afterburnenergy += ((1 - dam) * reactor.Capacity());
-        } else if (randnum >= cargovolume_damage_prob) {
-            CargoVolume *= dam;
-        } else if (randnum >= upgradevolume_damage_prob) {
-            UpgradeVolume *= dam;
-        } else if (randnum >= cargo_damage_prob) {
+        std::random_device dev;
+        std::mt19937 rng(dev());
+        std::uniform_int_distribution<std::mt19937::result_type> dist20(0,19); // distribution in range [1, 6]
+
+        switch(dist20(rng)) {
```

Comment:
Qs:

1. Can we get some documentation on this?
2. Did you mean to skip the `break` and allow fall through on cases 1-7?
3. Everything seems to add some damage; what about the default case? I see it commented-out - we should have some documentation on why it's not needed but even then it'd probably be good to have it and log it just in case.

Reply:
1. It's a simple algorithm to replace the complex one before. Random 20 (for now). 1-8 damages something by calling component->Damage();
2. **Yikes! Definitely not.** This is why we review stuff. The issue here of course is we can't really test this in the game. We need to pull this out of unit_generic and add a unit test to do it properly. 
3. It's a simple mechanism to ensure components don't get damaged every time we damage the hull.

Reply:
> 1. It's a simple algorithm to replace the complex one before. Random 20 (for now). 1-8 damages something by calling component->Damage();

more thinking of the next time we come to this code and wanting to understand it and why it's there; also per :two: and why the `break` should *not* be there so it doesn't get added accidentally by someone well-meaning in the future (e.g solving linter issues) so having some comments in the code to resolve that